### PR TITLE
Closes #8609: Intermittent test failues in WebExtensionSupportTest

### LINKED ISSUE #8609: Intermittently WebExtensionSupportTest reacts to new extension being installed
```
reacts to new extension being installed

Wanted but not invoked:
webExtension.registerActionHandler(
    Mock for EngineSession, hashCode: 464457386,
    <Capturing argument>
);
-> at mozilla.components.support.webextensions.WebExtensionSupportTest.reacts to new extension being installed(WebExtensionSupportTest.kt:330)

However, there were exactly 12 interactions with this mock:
webExtension.getId();
-> at mozilla.components.support.webextensions.WebExtensionSupport.registerInstalledExtension(WebExtensionSupport.kt:314)

webExtension.getId();
-> at mozilla.components.support.webextensions.WebExtensionSupport.toState(WebExtensionSupport.kt:451)

webExtension.getUrl();
-> at mozilla.components.support.webextensions.WebExtensionSupport.toState(WebExtensionSupport.kt:452)

webExtension.getMetadata();
-> at mozilla.components.support.webextensions.WebExtensionSupport.toState(WebExtensionSupport.kt:453)

webExtension.isEnabled();
-> at mozilla.components.support.webextensions.WebExtensionSupport.toState(WebExtensionSupport.kt:454)

webExtension.isAllowedInPrivateBrowsing();
-> at mozilla.components.support.webextensions.WebExtensionSupport.toState(WebExtensionSupport.kt:455)

webExtension.getId();
-> at mozilla.components.support.webextensions.WebExtensionSupportTest.reacts to new extension being installed(WebExtensionSupportTest.kt:320)

webExtension.getUrl();
-> at mozilla.components.support.webextensions.WebExtensionSupportTest.reacts to new extension being installed(WebExtensionSupportTest.kt:320)

webExtension.getMetadata();
-> at mozilla.components.support.webextensions.WebExtensionSupportTest.reacts to new extension being installed(WebExtensionSupportTest.kt:320)

webExtension.isEnabled();
-> at mozilla.components.support.webextensions.WebExtensionSupportTest.reacts to new extension being installed(WebExtensionSupportTest.kt:320)

webExtension.getId();
-> at mozilla.components.support.webextensions.WebExtensionSupportTest.reacts to new extension being installed(WebExtensionSupportTest.kt:323)

webExtension.getSupportActions();
-> at mozilla.components.support.webextensions.WebExtensionSupport.registerSessionHandlers(WebExtensionSupport.kt:405)


	at mozilla.components.support.webextensions.WebExtensionSupportTest.reacts to new extension being installed(WebExtensionSupportTest.kt:330)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.robolectric.RobolectricTestRunner$HelperTestRunner$1.evaluate(RobolectricTestRunner.java:601)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.robolectric.internal.SandboxTestRunner$2.evaluate(SandboxTestRunner.java:260)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:130)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:42)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.robolectric.internal.SandboxTestRunner$1.evaluate(SandboxTestRunner.java:84)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at androidx.test.ext.junit.runners.AndroidJUnit4.run(AndroidJUnit4.java:104)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:119)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
	at java.lang.Thread.run(Thread.java:748)


```

### PR DESCRIPTION
**Pull request by csadilek**
Left-over from the recent test refactoring. We need to make sure to not both verify and also rely on the state of store spies :).

### CODE REVIEW

File: components/support/webextensions/src/test/java/mozilla/components/support/webextensions/WebExtensionSupportTest.kt [hunk 1]
```diff
@@ -201,10 +202,8 @@ class WebExtensionSupportTest {
         val tabHandlerCaptor = argumentCaptor<TabHandler>()
         WebExtensionSupport.initialize(engine, store)
 
-        store.dispatch(EngineAction.LinkEngineSessionAction(tabId, engineSession)).joinBlocking()
```
Comment (Amejia481) [inline components/support/webextensions/src/test/java/mozilla/components/support/webextensions/WebExtensionSupportTest.kt:204]: We don't require the link actions anymore, right? :)

**Review by Amejia481 — APPROVED** [review]

File: components/support/webextensions/src/test/java/mozilla/components/support/webextensions/WebExtensionSupportTest.kt (hunk 1 above)
Comment (Amejia481) [reply components/support/webextensions/src/test/java/mozilla/components/support/webextensions/WebExtensionSupportTest.kt:204]: Is this related to the initial action patch in the store? :)

Comment (csadilek) [reply components/support/webextensions/src/test/java/mozilla/components/support/webextensions/WebExtensionSupportTest.kt:204]: @Amejia481 right, the engine session is now added directly when we set up the initial state in the test.

**Comment by csadilek** [comment]
run 1: 🌵

File: components/support/webextensions/src/test/java/mozilla/components/support/webextensions/WebExtensionSupportTest.kt (hunk 1 above)
Comment (Amejia481) [reply components/support/webextensions/src/test/java/mozilla/components/support/webextensions/WebExtensionSupportTest.kt:204]: 👍

**Comment by csadilek** [comment]
run 2: 🌳

**Comment by csadilek** [comment]
run 3: 🍏

**Comment by csadilek** [comment]
run 4: 🌲

**Comment by csadilek** [comment]
run 5: ✅ 🚢 🤞

**Pull request merged**